### DISCUSSION TITLE
ImmuDB usage on the long term. Can old data be truncated or removed?

### DISCUSSION THREAD

**Discussion by tutiplain** [codenotary/immudb#1375] [Q&A]
So, ImmuDB cannot delete data, or at least, you can still view old data with the time-travel feature, so that means that even delete transactions are recorded. What if I want to save disk space by removing unwanted time-travel data. Does ImmuDB provide a way in which time-travel data can be removed or truncated and just keep the most recent data? It may seem counterproductive because immutability is ImmuDB's flagship feature, but there has to be a way to save disk space if you are using ImmuDB in the long term, right? I read about the compactor tool, but I don't it removes the time-travel history, apparently it just rebuilds the indexes.

Is this a feature ImmuDB has or are there plans to add it?

**Comment by jeroiraz** [comment]
this is a really good point, physical data deletion or ways to deal with data that is not needed anymore is already on the plans for immudb. Meanwhile, a possible approach to deal with this problem is to create multiple databases. When a database is not needed anymore, it can be physically removed.

As you mentioned, index compaction is meant to reduce garbage on indexes that it's generated to to the append-only nature and snapshot isolation. But it's not currently deleting historical data.
